### ISSUE TITLE
[Button]: Missing focus styles on mobile

### ISSUE THREAD

**Issue by PlutaKatarzyna** [SAP/ui5-webcomponents#11024]
### Bug Description

By requirements, focus on Button component should be visible on desktop when click or focus by Tab key, and on mobile only by focus with Tab key.
However on mobile view, focus outline in Button component is not visible at all as `focused` attribute is not added.

<img width="458" alt="Image" src="https://github.com/user-attachments/assets/c7cefaef-380e-4c7f-bf37-54a6cf0b7819" />
<img width="458" alt="Image" src="https://github.com/user-attachments/assets/4f7c84da-3b00-44f5-add8-40aeddf182db" />

On desktop it works as expected.

### Affected Component

ui5-button

### Expected Behaviour

Focus outline is visible while navigating to the button using Tab key

### Isolated Example

https://sap.github.io/ui5-webcomponents/v1/components/Button/

### Steps to Reproduce

1. Open example with ui5-button
2. Open browser mobile preview - Toggle device toolbar
3. Use tab key to focus on button
4. Focus is not visiblre

### Priority

None

### UI5 Web Components Version

1.24.17

### Browser

Chrome

### Organization

SAP UDEx Components & Tokens

### Declaration

- [x] I’m not disclosing any internal or sensitive information.
